tokio/src/task/consume_budget.rs

```diff
@@ -0,0 +1,33 @@
+use std::future::Future;
+use std::pin::Pin;
+use std::task::{Context, Poll};
+
+/// Consumes a unit of budget and returns the execution back to the Tokio
+/// runtime *if* the task went out ouf budget.
+///
+/// The task will only yield if it ran out of its coop budget.
+/// It can be used in order to insert optional yield points into long
+/// computations that do not use Tokio resources like sockets or semaphores,
+/// without redundantly yielding to runtime each time.
+#[cfg_attr(docsrs, doc(cfg(feature = "rt")))]
+pub async fn consume_budget() {
```

Comment:
i wonder if we should pre-emptively mark this as an unstable feature, so that we can ship it sooner in the next release while reserving the ability to bikeshed the API for a bit longer?

Reply:
Could you point me to a code example showing how to mark a feature as unstable?

Reply:
nvm, just found https://github.com/tokio-rs/tokio/pull/4499, I'll follow it as a guideline

Reply:
If I view the generated docs, then it is not listed as unstable. You should add that to the `doc(cfg(..))`.
```Rust
#[cfg_attr(docsrs, doc(cfg(all(tokio_unstable, feature = "rt"))))]
```

Reply:
done, thanks!